pkg/controller/installlogmonitor/installlogmonitor_controller.go

```diff
@@ -0,0 +1,280 @@
+	)
+)
+
+func init() {
+	metrics.Registry.MustRegister(metricInstallErrors)
+}
+
+// Add creates a new InstallLogMonitro Controller and adds it to the Manager with default RBAC. The Manager will set fields on the Controller
+// and Start it when the Manager is Started.
+func Add(mgr manager.Manager) error {
+	return add(mgr, newReconciler(mgr))
+}
+
+// newReconciler returns a new reconcile.Reconciler
+func newReconciler(mgr manager.Manager) reconcile.Reconciler {
+	return &ReconcileInstallLog{Client: mgr.GetClient(), scheme: mgr.GetScheme()}
+}
+
+// add adds a new Controller to mgr with r as the reconcile.Reconciler
+func add(mgr manager.Manager, r reconcile.Reconciler) error {
+	// Create a new controller
+	c, err := controller.New("installlogmonitor-controller", mgr, controller.Options{Reconciler: r})
+	if err != nil {
+		return err
+	}
+
+	// Watch for changes to ConfigMap
+	err = c.Watch(&source.Kind{Type: &corev1.ConfigMap{}}, &handler.EnqueueRequestForObject{})
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+var _ reconcile.Reconciler = &ReconcileInstallLog{}
+
+// ReconcileInstallLog reconciles an install log ConfigMap object uploaded by the hive installmanager pod.
+type ReconcileInstallLog struct {
+	client.Client
+	scheme *runtime.Scheme
+}
+
+func (r *ReconcileInstallLog) isHiveInstallLog(cm *corev1.ConfigMap) (isInstallLog bool, needsMigration bool) {
+	if _, ok := cm.Labels[hivev1.HiveInstallLogLabel]; ok {
+		return true, false
+	}
+
+	// If it looks like one of our install logs that predates install manager images, return true
+	// and indicate it needs migration.
+	if _, ok := cm.Data["log"]; ok && strings.HasSuffix(cm.Name, "-install-log") {
+		return true, true
+	}
+
+	return false, false
+}
+
+// migrateInstallLog is a temporary function that will apply the new labels to pre-existing install log configmaps.
+func (r *ReconcileInstallLog) migrateInstallLog(cm *corev1.ConfigMap) error {
+	if cm.Labels == nil {
+		cm.Labels = map[string]string{}
+	}
+
+	cm.Labels[hivev1.HiveInstallLogLabel] = "true"
+
+	// We must guess at the cluster deployment name:
+	clusterDeploymentName := cm.Name[:len(cm.Name)-len("-install-log")]
+	cm.Labels[hivev1.HiveClusterDeploymentNameLabel] = clusterDeploymentName
+
+	return r.Client.Update(context.Background(), cm)
+}
+
+// Reconcile parses install log to monitor for known issues.
+// +kubebuilder:rbac:groups=core,resources=serviceaccounts;secrets;configmaps,verbs=get;list;watch;create;update;patch;delete
+func (r *ReconcileInstallLog) Reconcile(request reconcile.Request) (reconcile.Result, error) {
+	iLog := log.WithFields(log.Fields{
+		"configMap":  request.NamespacedName.String(),
+		"controller": controllerName,
+	})
+
+	// Load the regex configmap, if we don't have one, there's not much point proceeding here.
+	regexCM := &corev1.ConfigMap{}
+	err := r.Get(context.TODO(), types.NamespacedName{Name: regexConfigMapName, Namespace: hiveNamespace}, regexCM)
```

Comment:
Why not just mount this into the manager pod and point to it via env var? If you change the configmap the operator can redeploy the controller.

Reply:
ConfigMaps apparently don't have generations but the way this is setup right now, a new ConfigMap implies a new operator image which implies a new controller image, so it would get restarted automatically.

Is this better though? My understanding is because we're watching ConfigMaps it's already cached in memory, going to disk is only going to be slower isn't it?

Reply:
The thinking was that you'd compile your regexes only once during initialization, but I don't feel strongly about it. I'm ok with the way it is as well.